internal/pkg/testhelpers/chainclock.go

```diff
@@ -30,6 +31,11 @@ func (fcc *fakeChainClock) EpochAtTime(t time.Time) int64 {
 	return fcc.epoch
 }
 
+// StartTimeOfEpoch returns the current time
+func (fcc *fakeChainClock) StartTimeOfEpoch(e uint64) time.Time {
+	return fcc.Now()
+}
```

Comment:
This fake chain clock doesn't seem right at all. It should use a fake clock but a real genesis timestamp and calculations. In fact I would think a real chain clock (wrapping a fake clock) would be adequate for testing.

Reply:
I agree with this